sci-mathematics/gmp_ecm/gmp_ecm-7.0.4.recipe

```diff
@@ -0,0 +1,70 @@
+SUMMARY="Elliptic Curve Method for Integer Factorization"
+DESCRIPTION="C implementation of the Elliptic Curve Method for factorizing integers"
+HOMEPAGE="http://ecm.gforge.inria.fr/"
+COPYRIGHT="2001-2016 GMP-ECM Authors"
+LICENSE="
+	GNU GPL v3
+	GNU LGPL v3
+	"
+REVISION="1"
+SOURCE_URI="https://gforge.inria.fr/frs/download.php/file/36224/ecm-$portVersion.tar.gz"
+CHECKSUM_SHA256="0cf7b3eee8462cc6f98b418b47630e1eb6b3f4f8c3fc1fb005b08e2a1811ba43"
+SOURCE_DIR="ecm-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	gmp_ecm$secondaryArchSuffix = $portVersion
+	cmd:ecm$secondaryArchSuffix = $portVersion
+	lib:libecm$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	gmp_ecm${secondaryArchSuffix}_devel = $portVersion
+	devel:libecm$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	haiku$secondaryArchSuffix
+	lib:libecm$secondaryArchSuffix = $portVersion
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libgmp$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:cmp
+	cmd:make
+	cmd:m4
+	cmd:gcc$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	runConfigure ./configure --enable-shared --enable-openmp
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	install -Dm644 AUTHORS "$docDir/AUTHORS"
```

Comment:
@fbrosson suggests doing so

Reply:
Well, I had suggested this variant:
```sh
	install -d -m 755 "$docDir"
	install -t "$docDir" -m 644 AUTHORS
```
but I just noticed that gmp_ecm also has a few more files that we could include in $docDir: NEWS, README, README.gpu and README.lib. So maybe we could use this;
```sh
	install -d "$docDir"
	install -t "$docDir" AUTHORS NEWS README*
```
(Yes, we can even drop the `-m 755` and `-m 644` since the files already have the right permissions.)

Hmm, maybe we should move NEWS to the devel package, with:
```sh
	packageEntries devel \
		"$developDir" \
		"$docDir/NEWS"
```